socorro/lib/transaction.py

```diff
@@ -2,14 +2,22 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+"""
+This module contains convenience classes for managing "transactions"
+that involve doing something and then committing or rolling back
+depending on what happened.
+"""
+
 import sys
 import time
 
 from configman.config_manager import RequiredConfig
 from configman import Namespace
 from six import reraise
 
-from socorro.external.postgresql.dbapi2_util import DBApiUtilNonFatalBaseException
+
+class NonFatalException(Exception):
+    """Exception for a non-fatal transaction that doesn't require reconnecting"""
```

Comment:
We use the `TransactionExecutor` and friends across all the crash storage classes--not just the database. So I redid this part to undo tying it to our postgres stuff.